atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/DbKvsOracleExtension.java

```diff
@@ -24,60 +26,54 @@
 import com.palantir.atlasdb.keyvalue.dbkvs.impl.OverflowMigrationState;
 import com.palantir.atlasdb.keyvalue.dbkvs.impl.SimpleTimedSqlConnectionSupplier;
 import com.palantir.atlasdb.keyvalue.dbkvs.impl.SqlConnectionSupplier;
-import com.palantir.docker.compose.DockerComposeRule;
+import com.palantir.docker.compose.DockerComposeExtension;
 import com.palantir.docker.compose.connection.Container;
 import com.palantir.docker.compose.connection.DockerPort;
 import com.palantir.nexus.db.pool.ConnectionManager;
 import com.palantir.nexus.db.pool.ReentrantManagedConnectionSupplier;
 import com.palantir.nexus.db.pool.config.ConnectionConfig;
 import com.palantir.nexus.db.pool.config.ImmutableMaskedValue;
 import com.palantir.nexus.db.pool.config.OracleConnectionConfig;
+import java.io.IOException;
 import java.net.InetSocketAddress;
 import java.sql.Connection;
 import java.time.Duration;
 import java.util.concurrent.Callable;
 import org.awaitility.Awaitility;
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
-import org.junit.runner.RunWith;
-import org.junit.runners.Suite;
-import org.junit.runners.Suite.SuiteClasses;
-
-@RunWith(Suite.class)
-@SuiteClasses({
-    DbKvsOracleTargetedSweepIntegrationTest.class,
-    DbKvsOracleKeyValueServiceTest.class,
-    DbKvsOracleSerializableTransactionTest.class,
-    DbKvsOracleSweepTaskRunnerTest.class,
-    DbKvsOracleGetCandidateCellsForSweepingTest.class,
-    OverflowSequenceSupplierEteTest.class,
-    OracleTableNameMapperEteTest.class,
-    OracleEmbeddedDbTimestampBoundStoreTest.class,
-    OracleNamespaceDeleterIntegrationTest.class,
-    OracleAlterTableIntegrationTest.class
-})
-public final class DbKvsOracleTestSuite {
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public final class DbKvsOracleExtension implements BeforeAllCallback, ExtensionContext.Store.CloseableResource {
     private static final String LOCALHOST = "0.0.0.0";
     private static final int ORACLE_PORT_NUMBER = 1521;
 
-    private DbKvsOracleTestSuite() {
-        // Test suite
-    }
+    private static volatile boolean isInitialized = false;
 
-    @ClassRule
```

Comment:
Background info: Normally, even if class rules are static (shared among multiple test classes), their `beforeAll` and `afterAll` methods are called for each test class. For example, a `DockerComposeRule` instance is shared among multiple test classes, but `beforeAll` and `afterAll` are not called only once. Instead, they are called repeatedly for each class (which I find really error-prone and hard to reason about the code, but that's how things work, so we need to be extra careful when dealing with rule chains, etc.).

Now, the statements I wrote above also apply to abstract classes. So if abstract classes have a `ClassRule`, it is shared among child test classes, but `beforeAll` and `afterAll` are called separately for every child class.

The peculiar part is that if we are using a suite class, then `beforeAll` and `afterAll` are executed only once, even if there are multiple test classes. So in this case, the `beforeAll` and `afterAll` methods of `DockerComposeRule` are called only once (which saves us time). This is why, when we change this class to be an extension, we need to ensure that the `beforeAll` and `afterAll` of `DockerComposeRule` is called once.